src/test/java/net/snowflake/client/jdbc/ResultSetStructuredTypesLatestIT.java

```diff
@@ -0,0 +1,167 @@
+                  + "'timestamp_ntz': '2021-12-23 09:44:44'::TIMESTAMP_NTZ, "
+                  + "'timestamp_tz': '2021-12-24 09:45:45 +0800'::TIMESTAMP_TZ, "
+                  + "'date': '2023-12-24'::DATE, "
+                  + "'time': '12:34:56'::TIME, "
+                  + "'binary': TO_BINARY('616263', 'HEX'), "
+                  + "'simpleClass': {'string': 'b'}"
+                  + "}::OBJECT("
+                  + "string VARCHAR, "
+                  + "b TINYINT, "
+                  + "s SMALLINT, "
+                  + "i INTEGER, "
+                  + "l BIGINT, "
+                  + "f FLOAT, "
+                  + "d DOUBLE, "
+                  + "bd DOUBLE, "
+                  + "bool BOOLEAN, "
+                  + "timestamp_ltz TIMESTAMP_LTZ, "
+                  + "timestamp_ntz TIMESTAMP_NTZ, "
+                  + "timestamp_tz TIMESTAMP_TZ, "
+                  + "date DATE, "
+                  + "time TIME, "
+                  + "binary BINARY, "
+                  + "simpleClass OBJECT(string VARCHAR)"
+                  + ")"); ) {
+        resultSet.next();
+        AllTypesClass object = resultSet.getObject(1, AllTypesClass.class);
+        assertEquals("a", object.getString());
+        assertEquals(1, (long) object.getB());
+        assertEquals(2, (long) object.getS());
+        assertEquals(3, (long) object.getI());
+        assertEquals(4, (long) object.getL());
+        assertEquals(1.1, (double) object.getF(), 0.01);
+        assertEquals(2.2, (double) object.getD(), 0.01);
+        assertEquals(BigDecimal.valueOf(3.3), object.getBd());
+        assertEquals(
+            Timestamp.valueOf(LocalDateTime.of(2021, 12, 22, 9, 43, 44)), object.getTimestampLtz());
+        assertEquals(
+            Timestamp.valueOf(LocalDateTime.of(2021, 12, 23, 10, 44, 44)),
+            object.getTimestampNtz());
+        assertEquals(
+            Timestamp.valueOf(LocalDateTime.of(2021, 12, 24, 2, 45, 45)), object.getTimestampTz());
+        assertEquals(Date.valueOf(LocalDate.of(2023, 12, 24)), object.getDate());
+        assertEquals(Time.valueOf(LocalTime.of(12, 34, 56)), object.getTime());
+        assertArrayEquals(new byte[] {'a', 'b', 'c'}, object.getBinary());
+        assertTrue(object.getBool());
+        assertEquals("b", object.getSimpleClass().getString());
+      }
+    }
+  }
+
+  // TODO Structured types feature exists only on QA environments
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testMapStructsFromChunks() throws SQLException {
+    try (Connection connection = init();
+        Statement statement = connection.createStatement();
+        ResultSet resultSet =
+            statement.executeQuery(
+                "select {'string':'a'}::OBJECT(string VARCHAR) FROM TABLE(GENERATOR(ROWCOUNT=>30000))"); ) {
+      while (resultSet.next()) {
+        SimpleClass object = resultSet.getObject(1, SimpleClass.class);
```

Comment:
should we register the class also in this test?

Reply:
yes. We should test both scenarios